attach/CMakeLists.txt

```diff
@@ -5,5 +5,7 @@ if(CMAKE_BUILD_TYPE STREQUAL "Debug")
   endif()
 add_subdirectory(base_attach_impl)
 add_subdirectory(frida_uprobe_attach_impl)
-add_subdirectory(syscall_trace_attach_impl)
-add_subdirectory(text_segment_transformer)
+if(UNIX AND NOT APPLE)
+  add_subdirectory(syscall_trace_attach_impl)
+  add_subdirectory(text_segment_transformer)
+endif()
```

Comment:
I understand these are gonna fail for now due to x86. But is it a compilation error or a runtime error for now?

Reply:
A runtime error
- text segment transformer emits x86 assembly, these instructions are obviously illegal on aarch64

Reply:
> I understand these are gonna fail for now due to x86. But is it a compilation error or a runtime error for now?

A runtime error

text segment transformer emits x86 assembly now, and the corresponding trampoline code was designed for x86

Reply:
it fails during `cmake --build build` as well because headers are not defined and there are a lot of headers which will need to be changed.

Reply:
I asked it because, the way I see it, the syscall trace component is about supporting ARM overall, not an Apple-specific feature. We should support it eventually and that's orthogonal to this PR, which is about making libbpf optional.

Mac support will benefit from ARM support, but:
1. Linux ARM support also will
2. By removing libbpf, maybe we can get it working on Mac x86? Aka non-M chips and/or via rosetta

Reply:
Although PR started as just adding an option for disabling support for `libbpf`, I have tried to compile it on `aarch64` systems. We can create an issue which will remove this particular check by making it compatible for all systems but given our ultimate goal of making it run on macOS, I took one step in that direction, although for macOS systems with m* chip series in this case. Above check will disable `syscall_trace_attach_impl` and `text_segment_transformer` for APPLE systems.
As for rosetta systems, it is another setup on system and `brew` messes up architectures for packages, so don't wanna go through that. Maybe you can try it and let us know.